api/v1alpha1/loadbalancer_types.go

```diff
@@ -135,3 +142,39 @@ type SlowStart struct {
 	Window *metav1.Duration `json:"window"`
 	// TODO: Add support for non-linear traffic increases based on user usage.
 }
+
+// ZoneAware defines the configuration related to the distribution of requests between locality zones.
+type ZoneAware struct {
+	// PreferLocalZone configures zone-aware routing to prefer sending traffic to the local locality zone.
+	//
+	// +optional
+	// +notImplementedHide
+	PreferLocal *PreferLocalZone `json:"preferLocal,omitempty"`
+}
+
+// PreferLocalZone configures zone-aware routing to prefer sending traffic to the local locality zone.
+type PreferLocalZone struct {
+	// ForceLocalZone defines override configuration for forcing all traffic to stay within the local zone instead of the default behavior
+	// which maintains equal distribution among upstream endpoints while sending as much traffic as possible locally.
+	//
+	// +optional
+	// +notImplementedHide
+	Force *ForceLocalZone `json:"force,omitempty"`
+
+	// MinEndpointsThreshold is the minimum number of total upstream endpoints across all zones required to enable zone-aware routing.
+	//
+	// +optional
+	// +notImplementedHide
+	MinEndpointsThreshold *uint64 `json:"minEndpointsCount,omitempty"`
```

Comment:
is the uint type tied to envoy types ?

Reply:
Yes that maps to `min_cluster_size` and uses uint64 but can adjust to something else if preferred.

Ref: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/load_balancing_policies/common/v3/common.proto#extensions-load-balancing-policies-common-v3-localitylbconfig-zoneawarelbconfig